pages/hooks/useFetchNews.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from "@tanstack/react-query";
+import { QueryKeys } from "../queryKeys";
+
+const NEWS_API_KEY = "9bb1ce4f63d74db08d7433bd4bfd17a5";
+// Adjust the import path as necessary
+
+const NEWS_URL = `https://newsapi.org/v2/top-headlines?country=us&apiKey=${NEWS_API_KEY}`;
+
+async function getNews(): Promise<any> {
+  try {
+    const response = await fetch(NEWS_URL);
+    return response.json();
+  } catch (error) {
+    console.error("Error fetching news:", error);
+    return { error: "Failed to fetch news" };
+  }
```

Comment:
You should `try-catch` like this. ReactQuery would always return a success if you do. If you need to do stuff to the error, use the `onError` key that you can pass to `useQuery`